_studio/mfx_lib/shared/src/mfx_mpeg2_encode_vaapi.cpp

```diff
@@ -380,18 +387,74 @@ mfxStatus VAAPIEncoder::QueryEncodeCaps(ENCODE_CAPS & caps)
     caps.MbQpDataSupport   = 1;
-    caps.SkipFrame         = 1;
 
     caps.SliceIPBOnly      = 1;
-    caps.MaxNum_Reference  = 1;
-    caps.MaxPicWidth       = 1920;
-    caps.MaxPicHeight      = 1088;
     caps.NoInterlacedField = 0; // Enable interlaced encoding
-    caps.SliceStructure    = 1; // 1 - SliceDividerSnb; 2 - SliceDividerHsw; 3 - SliceDividerBluRay; the other - SliceDividerOneSlice
 
+    std::map<VAConfigAttribType, size_t> idx_map;
+    VAConfigAttribType attr_types[] = {
+        VAConfigAttribMaxPictureHeight,
+        VAConfigAttribMaxPictureWidth,
+        VAConfigAttribEncSkipFrame,
+        VAConfigAttribEncMaxRefFrames,
+        VAConfigAttribEncSliceStructure
+    };
+
+    std::vector<VAConfigAttrib> attrs;
+    attrs.reserve(sizeof(attr_types) / sizeof(attr_types[0]));
+
+    for (size_t i=0; i < sizeof(attr_types)/sizeof(attr_types[0]); ++i)
+    {
+        attrs.push_back( createVAConfigAttr(attr_types[i], 0) );
+        idx_map[ attr_types[i] ] = i;
+    }
+
+    VAEntrypoint entrypoint = VAEntrypointEncSlice;
+
+    if(codecProfileType == 0)
+        codecProfileType = m_initCodecProfile;
+
+    VAStatus vaSts = vaGetConfigAttributes(m_vaDisplay,
+                          ConvertProfileTypeMFX2VAAPI(codecProfileType),
+                          entrypoint,
+                          attrs.data(), attrs.size());
+
+    MFX_CHECK(!(VA_STATUS_ERROR_UNSUPPORTED_ENTRYPOINT == vaSts ||
+                VA_STATUS_ERROR_UNSUPPORTED_PROFILE    == vaSts),
+                MFX_ERR_UNSUPPORTED);
+
+    if ((attrs[idx_map[VAConfigAttribMaxPictureWidth]].value != VA_ATTRIB_NOT_SUPPORTED) &&
+        (attrs[idx_map[VAConfigAttribMaxPictureWidth]].value != 0))
+        caps.MaxPicWidth  = attrs[idx_map[VAConfigAttribMaxPictureWidth]].value;
+    else
+        caps.MaxPicWidth = 1920;
+
+    if ((attrs[idx_map[VAConfigAttribMaxPictureHeight]].value != VA_ATTRIB_NOT_SUPPORTED) &&
+        (attrs[idx_map[VAConfigAttribMaxPictureHeight]].value != 0))
+        caps.MaxPicHeight = attrs[idx_map[VAConfigAttribMaxPictureHeight]].value;
+    else
+        caps.MaxPicHeight = 1920;
+
+    if ((attrs[idx_map[VAConfigAttribEncSkipFrame]].value != VA_ATTRIB_NOT_SUPPORTED) &&
+        (attrs[idx_map[VAConfigAttribEncSkipFrame]].value != 0))
+        caps.SkipFrame  = attrs[idx_map[VAConfigAttribEncSkipFrame]].value;
+    else
+        caps.SkipFrame = 1;
+
+    if ((attrs[idx_map[VAConfigAttribEncMaxRefFrames]].value != VA_ATTRIB_NOT_SUPPORTED) &&
+        (attrs[idx_map[VAConfigAttribEncMaxRefFrames]].value != 0))
+        caps.MaxNum_Reference  = attrs[idx_map[VAConfigAttribEncMaxRefFrames]].value;
+    else
+        caps.MaxNum_Reference = 1;
+
+    if ((attrs[idx_map[VAConfigAttribEncSliceStructure]].value != VA_ATTRIB_NOT_SUPPORTED) &&
+        (attrs[idx_map[VAConfigAttribEncSliceStructure]].value != 0))
+        caps.SliceStructure  = attrs[idx_map[VAConfigAttribEncSliceStructure]].value;
+    else
+        caps.SliceStructure = 2;
```

Comment:
it was '1' previously. why did you change to '2'?

Reply:
And what 2 actually means? Why we don't have some enum list with self speaking names?!

Reply:
@dmitryermilov  yes, it was. But driver returns SliceStructure = 2. What value should we use in this case?
@dvrogozh  SliceStructure = 2 means SliceDividerHsw, but I can't find any enum list for this parameter

Reply:
> @dmitryermilov yes, it was. But driver returns SliceStructure = 2. What value should we use in this case?

'else' block for the case when driver doesn't return caps. I don't mind changing default value but it's a subject for separate commit

> @dmitryermilov yes, it was. But driver returns SliceStructure = 2. What value should we use in this case?
see https://github.com/Intel-Media-SDK/MediaSDK/pull/1814/files#diff-21db52ca9c332ae4833c901c5be556ea